File: 1666-make-the-string-great/make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        string stack;
        
        for (char ch : s) {
            // If the top of the stack is the "opposite case" of ch, they cancel out
            if (!stack.empty() && stack.back() != ch && 
                tolower(stack.back()) == tolower(ch)) {
                stack.pop_back();
            } else {
                stack.push_back(ch);
            }
        }
        
        return stack;
    }
};

File: 1666-make-the-string-great/make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []
        
        for ch in s:
            # If the top of the stack is the "opposite case" of ch, they cancel out
            if stack and stack[-1] != ch and stack[-1].lower() == ch.lower():
                stack.pop()
            else:
                stack.append(ch)
        
        return ''.join(stack)